dhall/src/Dhall/Binary.hs

```diff
@@ -934,7 +941,7 @@ strip55799Tag term =
             TDouble a
 
 -- | Encode a Dhall expression as a CBOR `Term`
-encodeExpression :: Expr s Import -> Term
+encodeExpression :: Expr X Import -> Term
```

Comment:
How about specialising to `Src` instead and changing the definition to `encode . denote`?

```suggestion
encodeExpression :: Expr Src Import -> Term
```

This function could then be used where we currently have to explicitly `denote`.

Reply:
Would it be possible to either (A) keep the polymorphic type by including a `denote` within this function or (B) delete this function?  The context is that this function only exists for backwards compatibility, so if we're going to break backwards compatibility then there's no reason to keep this function around (when users can use `encode` directly)

Reply:
I went with (A)